File: server/src/knowledge/ingestors/pdfIngestor.js
import { BaseIngestor } from '../baseIngestor.js';
import { parseDocument, chunkText } from '../../rag.js';

// Real implementation — wraps the existing parseDocument/chunkText pair used
// by /api/upload today, normalized to the BaseIngestor contract.
export class PdfIngestor extends BaseIngestor {
  constructor() {
    super('document');
  }

  async ingest(source) {
    const text = await parseDocument(source.filePath, 'application/pdf');
    const chunks = chunkText(text);
    // pdf-parse does not give us per-page boundaries for free; page_number
    // stays null until a page-aware parser replaces pdf-parse.
    return chunks.map(content => ({ content, metadata: { page_number: null } }));
  }
}

File: server/app/knowledge/ingestors/pdf_ingestor.py
from ..base_ingestor import BaseIngestor
from ...rag import parse_document, chunk_text


class PdfIngestor(BaseIngestor):
    """Wraps the existing parse_document/chunk_text pair used by /api/upload
    today, normalized to the BaseIngestor contract."""

    def __init__(self):
        super().__init__("document")

    async def ingest(self, source: dict) -> list[dict]:
        parsed = await parse_document(source["filePath"], "application/pdf")
        chunks = chunk_text(parsed["text"])
        return [{"content": c, "metadata": {"page_number": None}} for c in chunks]
